pkg/descheduler/framework/plugins/loadaware/utilization_util.go

```diff
@@ -285,22 +419,54 @@ func evictPodsFromSourceNodes(
 			}),
 		)
 		klog.V(4).InfoS("Evicting pods from node",
-			"nodePool", nodePoolName, "node", klog.KObj(srcNode.node), "usage", srcNode.usage,
-			"allPods", len(srcNode.allPods), "nonRemovablePods", len(nonRemovablePods), "removablePods", len(removablePods))
+			"nodePool", nodePoolName, "node", klog.KObj(srcNode.node), "prod", prod, "usage", srcNode.usage,
+			"allPods", len(allPods), "nonRemovablePods", len(nonRemovablePods), "removablePods", len(removablePods))
 
 		if len(removablePods) == 0 {
 			klog.V(4).InfoS("No removable pods on node, try next node", "node", klog.KObj(srcNode.node), "nodePool", nodePoolName)
 			continue
 		}
 		sortPodsOnOneOverloadedNode(srcNode, removablePods, resourceWeights)
-		evictPods(ctx, nodePoolName, dryRun, removablePods, srcNode, totalAvailableUsages, podEvictor, podFilter, continueEviction, evictionReasonGenerator)
+		evictPods(ctx, nodePoolName, dryRun, prod, removablePods, srcNode, totalAvailableUsages, podEvictor, podFilter, continueEviction, evictionReasonGenerator)
 	}
 }
 
+func targetAvailableUsage(destinationNodes []NodeInfo, resourceNames []corev1.ResourceName, prod bool) (map[corev1.ResourceName]*resource.Quantity, []*corev1.Node) {
+	var targetNodes []*corev1.Node
+	totalAvailableUsages := map[corev1.ResourceName]*resource.Quantity{}
+	for _, destinationNode := range destinationNodes {
+		targetNodes = append(targetNodes, destinationNode.node)
+		for _, resourceName := range resourceNames {
+			quantity, ok := totalAvailableUsages[resourceName]
+			if !ok {
+				switch resourceName {
+				case corev1.ResourceCPU:
+					quantity = resource.NewMilliQuantity(0, resource.DecimalSI)
+				case corev1.ResourceMemory, corev1.ResourceEphemeralStorage, corev1.ResourceStorage:
+					quantity = resource.NewQuantity(0, resource.BinarySI)
+				default:
+					quantity = resource.NewQuantity(0, resource.DecimalSI)
+				}
+				totalAvailableUsages[resourceName] = quantity
+			}
+			if prod {
+				quantity.Add(*destinationNode.thresholds.prodHighResourceThreshold[resourceName])
+				quantity.Sub(*destinationNode.prodUsage[resourceName])
```

Comment:
The prod availble usage should be `min(prodHighThreshold - prodUsage, totalHighThreshold - totalUsage).` Otherwise a node may be overutilized on a total-dimension after prod pod migration.

Reply:
The corresponding calculation method has been updated.